packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.spec.js

```diff
@@ -1,60 +1,52 @@
+import { render, screen } from '@testing-library/react'
 import asyncStates from '@zooniverse/async-states'
-import { shallow } from 'enzyme'
-
 import { SubjectViewer } from './SubjectViewer'
-import SingleImageViewer from './components/SingleImageViewer'
-import JSONDataViewer from './components/JSONDataViewer'
 
 describe('Component > SubjectViewer', function () {
   it('should render without crashing', function () {
-    const wrapper = shallow(<SubjectViewer />)
-    expect(wrapper).to.be.ok()
+    render(<SubjectViewer />)
+    expect(screen).to.be.ok()
   })
 
   it('should render nothing if the subject store is initialized', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.initialized} />)
-    expect(wrapper.type()).to.be.null()
+    const { container } = render(<SubjectViewer subjectQueueState={asyncStates.initialized} />)
+    expect(container.firstChild).to.be.null()
   })
-
+  
   it('should render a loading indicator if the subject store is loading', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.loading} />)
-    expect(wrapper.text()).to.equal('SubjectViewer.loading')
+    render(<SubjectViewer subjectQueueState={asyncStates.loading} />)
+    expect(screen.getByText('SubjectViewer.loading')).to.exist()
   })
 
   it('should render nothing if the subject store errors', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.error} />)
-    expect(wrapper.type()).to.be.null()
+    const { container } = render(<SubjectViewer subjectQueueState={asyncStates.error} />)
+    expect(container.firstChild).to.be.null()
   })
 
-  it('should render a subject viewer if the subject store successfully loads', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.success} subject={{ viewer: 'singleImage' }} />)
-    expect(wrapper.find(SingleImageViewer)).to.have.lengthOf(1)
-  })
-
-  it('should pass along the viewer configuration', function () {
-    const viewerConfiguration = {
-      zoomConfiguration: {
-        direction: 'both',
-        minZoom: 1,
-        maxZoom: 10,
-        zoomInValue: 1.2,
-        zoomOutValue: 0.8
-      }
-    }
-
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.success} subject={{ viewer: 'variableStar', viewerConfiguration }} />)
-    expect(wrapper.find(JSONDataViewer).props().viewerConfiguration).to.deep.equal(viewerConfiguration)
+  it('should render the VolumetricViewer if the subject store successfully loads', function () {
```

Comment:
```suggestion
  it('should render the VolumetricViewer if isVolumetric = true', function () {
```

This test is specific to a volumetric project. You've deleted the generic unit test for 'if the subject store successfully loads'.

Reply:
Yes - this was because the spec was originally in Enzyme and mounting/rendering sub-components gets pretty dicy when converting to RTL... So, my thought was to make a more "robust" test of the SubjectViewer that handles the async nature of the VolumetricViewer because, if it can handle that, it should be able to render the default Viewer. I admit it compromises completeness of test coverage (handling the rendering of the non-async viewers), but the test I added is meant to make up for that with it being the more "complex" scenario due to its async nature.